### PR TITLE
[ENTESB-10817] Openshift 4.1 + FMP: Unexpected response (403 Forbidden), to the authorization request. Missing header:[Location]!

### PR DESCRIPTION
**Pull request by gastaldi**
Closes #1641

PS: Sorry about the formatting. I don't know which formatter is used

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

File: openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java [hunk 1]
```diff
@@ -16,124 +16,130 @@
...
-        if (token == null || token.isEmpty()) {
-          throw new IOException("Unexpected response(" + response.code() + " " + response.message() + "), to the authorization request. Missing header:[" + LOCATION + "]!");
-        }
-        token = token.substring(token.indexOf(BEFORE_TOKEN) + BEFORE_TOKEN.length());
-        token = token.substring(0, token.indexOf(AFTER_TOKEN));
-        return token;
+      String body = response.body().string();
+      JsonNode jsonResponse = Serialization.jsonMapper().readTree(body);
+      String authorizationServer = jsonResponse.get("authorization_endpoint").asText();
+      response.body().close();
```
Comment (rohanKanojia) [inline openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java]: Doesn't using `response.body().string()` closes body itself?

File: openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java [hunk 2]
```diff
@@ -16,124 +16,130 @@
...
+      token = config.getOauthToken();
+      oauthToken.set(token);
+    }
 
 
-      //If token was obtain, then retry request using the obtained token.
-      if (Utils.isNotNullOrEmpty(token)) {
-        // Close the previous response to prevent leaked connections.
-        response.body().close();
+    //If token was obtain, then retry request using the obtained token.
```
Comment (rohanKanojia) [inline openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java]: I know you didn't make this change; might be wrong as I'm not native English speaker, but shouldn't it be obtained here?

File: openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java (hunk 1 above)
Comment (gastaldi) [reply openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java]: Not sure, but I know that `response.close()` does

**Comment by gastaldi** [comment]
Yes, I tested this and works fine

File: openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java (hunk 2 above)
Comment (gastaldi) [reply openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java]: Yes, You're right. Fixing that

**Comment by rohanKanojia** [comment]
oh, okay. Then it must be fine

**Comment by rohanKanojia** [comment]
@gastaldi : Could you please add a line to changelog also ;-)

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**